graphs/gen/batagelj_brandes.go

```diff
@@ -0,0 +1,355 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The functions in this file are random graph generators from the paper
+// by Batagelj and Brandes http://algo.uni-konstanz.de/publications/bb-eglrn-05.pdf
+
+package gen
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	"github.com/gonum/graph"
+	"github.com/gonum/graph/simple"
+)
+
+// Gnp constructs a graph in the destination, dst, of order n. Edges between nodes
+// are formed with the probability, p. If src is not nil it is used as the random source
+// otherwise rand.Float64 is used. The graph is constructed in O(n+m) time.
+func Gnp(dst GraphBuilder, n int, p float64, src *rand.Rand) error {
+	if p == 0 {
+		return nil
+	}
+	if p < 0 || p > 1 {
+		return fmt.Errorf("gen: bad probability: p=%v", p)
```

Comment:
We already have errors here, so I think I'll keep the error. The package is really only intended to provide graphs for examples, tests and benchmarks, so I don't think we need to be as hard as usual.